### PR TITLE
The Exorcism

### LINKED ISSUE #471: JetSelector: m_useCutFlow being ignored in initialize
Since [a condition involving](https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/JetSelector.cxx#L350) `m_useCutFlow` appears before [the configuration of the algorithm](https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/JetSelector.cxx#L377), the value of `m_useCutFlow` given in the configuration file is ignored. As `m_useCutFlow` [is set to true in the constructor](https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/JetSelector.cxx#L67), the block following the condition will always be executed, resulting in abortion of the algorithm [if `cutflow` is not available](https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/JetSelector.cxx#L354).

### PR DESCRIPTION
**Pull request by kratsg**
> The path of the righteous configuration is beset on all sides by the iniquities of the selfish and the tyranny of TE(n)vil files.
> 
> Blessed is he who, in the name of code quality and open-source, shepherds the weak through the valley of ROOTness, for he is truly his brother's keeper and the finder of lost programmers.
> 
> And I will force push upon thee with great vengeance and furious anger those who attempt to commit and destroy our framework. And you will know my name is the Git when I lay my vengeance upon thee.
> 

tl;dr - no more TEnv files. This will resolve #471 indirectly (issues raised about confusing people and stuff).

> The power of Linus compels you!

### CODE REVIEW

**Comment by johnda102** [comment]
looks good.

Minor thing we might was to keep in BasicEventSelection:

// if truth level make sure parameters are set prope
- if( m_truthLevelOnly ) {
- Info("configure()", "Truth only! Turn off trigger stuff");
- m_triggerSelection = "";
- m_applyTriggerCut = m_storeTrigDecisions = m_storePassL1 = m_storePassHLT
= m_storeTrigKeys = false;
- Info("configure()", "Truth only! Turn off GRL");
- m_applyGRLCut = false;
- Info("configure()", "Truth only! Turn off Pile-up Reweight");
-m_doPUreweighting = false;
}

On Sat, Feb 6, 2016 at 6:32 PM, xAH Bot <notifications@github.com> wrote:
>
> Everything checked out ok, to merge.
>
> Commit sha1: 1c0946b
>
> —

> Reply to this email directly or view it on GitHub
> <https://github.com/UCATLAS/xAODAnaHelpers/pull/476#issuecomment-180816048>
> .
>



-- 
University of Chicago
http://hep.uchicago.edu/~johnda

**Comment by kratsg** [comment]
> Minor thing we might was to keep in BasicEventSelection:

It's funny you say that -- because that's only ever called if you provide a TEnv config file. Otherwise, it was never used.... I can add it back in.

**Comment by johnda102** [comment]
Can we merge here ?

**Comment by kratsg** [comment]
I haven't added the minor thing in yet.

**Comment by johnda102** [comment]
ok doing it now

On Thu, Feb 11, 2016 at 8:44 AM, Giordon Stark <notifications@github.com>
wrote:

> I haven't added the minor thing in yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UCATLAS/xAODAnaHelpers/pull/476#issuecomment-182751126>
> .
>



-- 
University of Chicago
http://hep.uchicago.edu/~johnda
